verification/src/http_server/handlers/verification/solidity/handlers.rs

```diff
@@ -0,0 +1,106 @@
+use crate::{
+    compiler::{CompilerVersion, Compilers, CompilersError, Fetcher},
+    solidity::{VerificationSuccess, Verifier, VerifierInitializationError},
+    VerificationResponse, VerificationResult,
+};
+use actix_web::error;
+use ethers_solc::{
+    artifacts::{BytecodeHash, SettingsMetadata},
+    CompilerInput,
+};
+use std::{
+    fmt::{Debug, Display},
+    str::FromStr,
+};
+use thiserror::Error;
+
+const BYTECODE_HASHES: [BytecodeHash; 3] =
+    [BytecodeHash::Ipfs, BytecodeHash::None, BytecodeHash::Bzzr1];
+
+pub struct CompileAndVerifyInput<'a> {
+    pub compiler_version: CompilerVersion,
+    pub compiler_input: CompilerInput,
+    pub creation_tx_input: &'a str,
+    pub deployed_bytecode: &'a str,
+}
+
+#[derive(Error, Debug)]
+enum CompileAndVerifyError {
+    #[error("{0:#}")]
+    VerifierInitialization(#[from] VerifierInitializationError),
+    #[error("{0:#}")]
+    Compilation(#[from] CompilersError),
+    #[error("No contract could be verified with provided data")]
+    NoMatchingContracts,
+}
+
+pub(crate) async fn compile_and_verify_handler<T: Fetcher>(
```

Comment:
just a random idea, may be make this (and other functions) as `impl CompileAndVerifyInput`?